azure/terraform/diagnosticsettings.rego

```diff
@@ -0,0 +1,371 @@
+package rule
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_diagnostic_setting
+#
+# PR-AZR-0017-TRF
+#
+
+default log_keyvault = null
+
+azure_attribute_absence["log_keyvault"] {
+    resource := input.resources[_]
+    lower(resource.type) == "azurerm_monitor_diagnostic_setting"
+    not resource.properties.log
+}
+
+# providers/Microsoft.KeyVault
+azure_issue["log_keyvault"] {
+    resource := input.resources[_]
+    lower(resource.type) == "azurerm_monitor_diagnostic_setting"
+    not contains(lower(resource.properties.target_resource_id), "microsoft.keyvault/vaults") 
```

Comment:
@rezoan the diagnostics setting resource type is a general resource that we can use for any resource type in the Azure.
does this rule check specifically for the `keyvault` diagnostics settings?
for example, if we enable diagnostics setting for an application gateway, this rule fails?

Reply:
@farchide added checking for parent resource existence as discussed.